src/android/InAppBrowserDriver.java

```diff
@@ -0,0 +1,543 @@
+     */
+    public Boolean onReceivedHttpAuthRequest(CordovaWebView view, ICordovaHttpAuthHandler handler, String host, String realm) {
+        // Check if there is some plugin which can resolve this auth challenge
+        PluginManager pluginManager = null;
+        try {
+            pluginManager = (PluginManager)InAppBrowserImpl.invokeOn(inAppBrowser.getWebView().getView(), "getPluginManager", new Class[0], new Object[0]);
+        } catch (InAppBrowserImpl.InvokeException e) {
+        }
+
+        if (pluginManager != null) {
+            return pluginManager.onReceivedHttpAuthRequest(view, handler, host, realm);
+        } else {
+            return false;
+        }
+    }
+
+    /**
+     * Notify the host application that the InAppBrowserImpl is exiting.
+     *
+     * @return              Object to stop propagation or null
+     */
+    public Object onExit() {
+        closeDialog();
+        return new Object();
+    }
+
+    /**
+     * Notify the host application that the internal onFinish callback was called
+     *
+     * @param message       The string the callback was called with
+     * @param callbackId    The callback ID to send the result to
+     */
+    public void onCallbackFinish(String message, String callbackId) {
+        PluginResult scriptResult;
+
+        if (message == null || message.length() == 0) {
+            scriptResult = new PluginResult(PluginResult.Status.OK, new JSONArray());
+        } else {
+            try {
+                scriptResult = new PluginResult(PluginResult.Status.OK, new JSONArray(message));
+            } catch(JSONException e) {
+                scriptResult = new PluginResult(PluginResult.Status.JSON_EXCEPTION, e.getMessage());
+            }
+        }
+        inAppBrowser.getWebView().sendPluginResult(scriptResult, callbackId);
+    }
+
+    /**
+     * Inject an object (script or style) into the InAppBrowserImpl WebView.
+     *
+     * This is a helper method for the inject{Script|Style}{Code|File} API calls, which
+     * provides a consistent method for injecting JavaScript code into the document.
+     *
+     * If a wrapper string is supplied, then the source string will be JSON-encoded (adding
+     * quotes) and wrapped using string formatting. (The wrapper string should have a single
+     * '%s' marker)
+     *
+     * @param source      The source object (filename or script/style text) to inject into
+     *                    the document.
+     * @param jsWrapper   A JavaScript string to wrap the source string in, so that the object
+     *                    is properly injected, or null if the source string is JavaScript text
+     *                    which should be executed directly.
+     */
+    public void injectDeferredObject(String source, String jsWrapper) {
+        String scriptToInject;
+        if (jsWrapper != null) {
+            org.json.JSONArray jsonEsc = new org.json.JSONArray();
```

Comment:
:+1: Copy/paste didn't cause Android Studio to import it and not use the fully qualified class name.